js/app/controllers/calcontroller.js

```diff
@@ -140,17 +141,27 @@ app.controller('CalController', ['$scope', 'Calendar', 'CalendarService', 'VEven
 				$scope.$apply();
 			});
 		} else {
-			$scope.calendarsPromise = CalendarService.getPublicCalendar(constants.publicSharingToken).then(function(calendar) {
-				$scope.calendars = [calendar];
+			$scope.calendars = [];
+			var tokens = new Set(constants.publicSharingToken.split("."));
```

Comment:
Just use an array here please. `new Set[array]` has no support in Internet explorer: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set